src/go/configinfo/service_info.go

```diff
@@ -401,8 +415,9 @@ func addHttpRule(method *MethodInfo, r *annotationspb.HttpRule, addedRouteMatchW
 	}
 
 	httpRule := &httppattern.Pattern{
-		HttpMethod:  httpMethod,
-		UriTemplate: uriTemplate,
+		HttpMethod:     httpMethod,
+		UriTemplate:    uriTemplate,
+		UseHTTPBackend: !IsDiscoveryAPI(method.Operation()) && s.Options.TestOnlyHTTPBackendAddress != "",
```

Comment:
I think we should not use "useHTTPBackend" in httpRule.Pattern.  It is overloaded.

Can we just have a "fromHTTPRule" to indicate it is generated from http.rule,  not from RPC methods?

Which backend it should use can be dealt at routeGenerate time?  I think we should have enough info for that.

maybe we don't even need this

 we can say:  `POST /service_name/method_name` is a grpc method,  the others are HTTP.

Reply:
This is a good idea. Indeed I found we should be able to use path to know it's GRPC. And we don't need to add field in httpRule.Pattern. PTAL.